### PR TITLE
[async] switch to using jetty async instead of our own threadpool

### PR DESCRIPTION
**Pull request by nwadams**
switch to using jetty async instead of our own threadpool for http requests. 

Streamline calls

Update to beta2

### CODE REVIEW

File: berbix-java/src/main/java/com/berbix/BerbixApi.java [hunk 1]
```diff
@@ -139,153 +127,201 @@ private Tokens createTokens(FetchTokensResponse fetchTokensResponse) {
...
-      throw new IOException(e.getMessage());
-    }
+      request.send(new CompletableFutureResponseListener(completableFuture));
+
+      return completableFuture.thenApply(response -> {
+        if (response.getStatus() == 204 && responseClass == String.class) {
+          // cast string as String so T compiles.
+          return responseClass.cast("finished");
+        } else {
+          String responseData = response.getContentAsString();
```
Comment (ChristianHansen) [inline berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: How does a caller of public methods that call `tokenAuthRequest()` detect if a non-2XX code was returned e.g. if they have a bad API secret?

Comment (nwadams) [reply berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: They will get an exception they need to handle from. https://github.com/berbix/berbix-java/pull/2/files#diff-2cee81cb9fe60f33dcf4e0de932378d6639372e2a83b6184ba3c42888f202567R315

Comment (ChristianHansen) [reply berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: Can you point me at the docs for what status codes are considered failures? I skimmed the documentation, but I couldn't figure out what exactly constitutes "failures"

Comment (nwadams) [reply berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: It just checks whether there is a request exception or a response exception. The documentation isn't super clear at all. https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/client/api/Result.html#isFailed()

From what I could tell, 4xx and 5xx both were considered `failures` and 2xx was considered success. I didn't have any 3xx, but I imagine they would be failures too.

Comment (ChristianHansen) [reply berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: Got it, mind leaving a comment somewhere with you findings?

**Review by ChristianHansen — APPROVED** [review]

File: berbix-java/src/main/java/com/berbix/BerbixApi.java (hunk 1 above)
Comment (nwadams) [reply berbix-java/src/main/java/com/berbix/BerbixApi.java:215]: done

**Pull request merged**